storage/netstore.go

```diff
@@ -99,6 +100,7 @@ type NetStore struct {
 	requestGroup singleflight.Group
 	RemoteGet    RemoteGetFunc
 	logger       log.Logger
+	prod         *prod.Prod
```

Comment:
No, the dependency is the  other way round more natural.
Here  i would   assign a function or if you insist, an interface with      
`Recover` method

Reply:
this was changed to be similar to Deliver hook.
This was one of the reasons there was an import cycle error.